apm-lambda-extension/logsapi/subscribe_test.go

```diff
@@ -0,0 +1,148 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package logsapi
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSubscribeWithSamLocalEnv(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	os.Setenv("AWS_SAM_LOCAL", "true")
+	t.Cleanup(func() {
+		os.Unsetenv("AWS_SAM_LOCAL")
+	})
+	out := make(chan LogEvent)
+
+	err := Subscribe(ctx, "testID", []EventType{Platform}, out)
+	assert.Error(t, err)
+}
+
+func TestSubscribeAwsRequest(t *testing.T) {
+	listenerHost = "localhost"
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	out := make(chan LogEvent)
+	// For subscription request
+	expectedTypes := []EventType{Platform}
+	expectedBufferingCfg := BufferingCfg{
+		MaxItems:  10000,
+		MaxBytes:  262144,
+		TimeoutMS: 25,
+	}
+
+	// Create aws runtime API server and handler
+	awsRuntimeApiServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		request_bytes, _ := ioutil.ReadAll(r.Body)
```

Comment:
Some general naming things in this area of code, but also another opportunity to use `json.NewDecoder`